controllers/issuer_controller.go

```diff
@@ -91,31 +99,90 @@ func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
-			log.Info("Initializing the Issuer signer failed", "issuer", req.Name, "error", err)
-			issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", err.Error())
-			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		if issuerSpec.SelfSignCertificate != nil && *issuerSpec.SelfSignCertificate {
+			log.Info("Adding new signer for", "issuer", req.Name)
+			if s, err = r.KeyProvider.AddSigner(signerName, true); err != nil {
+				log.Info("Initializing the Issuer signer failed", "issuer", req.Name, "error", err)
+				issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", err.Error())
+				return ctrl.Result{Requeue: false}, nil
+			}
+		} else {
+			qa := &tcsapi.QuoteAttestation{}
+			qaReq := req.NamespacedName
+			if qaReq.Namespace == "" {
+				qaReq.Namespace = k8sutil.GetNamespace()
+			}
+			if err := r.Get(ctx, qaReq, qa); err != nil {
+				if apierrors.IsNotFound(err) {
+					// means no quoteattestation object, create new one
+					quote, publickey, err := r.KeyProvider.GetQuoteAndPublicKey()
+					if err != nil {
+						log.Info("Error preparing SGX quote", "error", err)
+						issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", fmt.Sprintf("failed to get sgx quote: %v", err.Error()))
+						return ctrl.Result{Requeue: true}, nil
+					}
+					ownerRef := &metav1.OwnerReference{
+						APIVersion: tcsapi.GroupVersion.String(),
+						Kind:       r.Kind,
+						Name:       issuer.GetName(),
+						UID:        issuer.GetUID(),
+					}
+					log.Info("Initiating quote attestation", "signer", signerName)
+					if err := k8sutil.QuoteAttestationDeliver(ctx, r.Client, qaReq, tcsapi.RequestTypeKeyProvisioning, signerName, quote, publickey, "", ownerRef); err != nil {
+						log.Error(err, "Error while createing quote attestation")
+						issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", fmt.Sprintf("failed to initiate quote attestation: %v", err.Error()))
+						return ctrl.Result{Requeue: true}, nil
+					}
+				} else {
+					log.Error(err, "Error while checking if quote attestation exists")
+					issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", fmt.Sprintf("failed to get quote attestation status: %v", err.Error()))
+					return ctrl.Result{Requeue: true}, nil
+				}
+			}
+
+			status := qa.Status.GetCondition(tcsapi.ConditionReady)
+			if status == nil || status.Status == v1.ConditionUnknown {
+				// Still not ready, retry later
+				issuerStatus.SetCondition(tcsapi.IssuerConditionReady, v1.ConditionFalse, "Reconcile", "waiting for key provisioning")
+				return ctrl.Result{Requeue: true}, nil
+			}
+
+			// Remove attestation object as we got the results
+			defer k8sutil.QuoteAttestationDelete(context.Background(), r.Client, qaReq)
+
+			if status.Status == v1.ConditionFalse {
```

Comment:
> 2. Attestation fails because the attestation endpoint is temporarily down, network connectivity is flaky, or the pods are scheduled in "wrong" order. This we will eventually recover from -> eventual consistency.

IMHO, all these cases could be handled at the attestation-controller side. It shall not update the CR status unless it got a reply from the server, else it should keep retrying without updating the CR status. Currently, we designed the QuoteAttestation.Status as a boolean state.

Reply:
Ok! So "false" means that the attestation failed because the quote/pub key was incorrect. For all other cases the attestation controller keeps trying. Then there is no prob here.